src/control/cmd/dmg/network.go

```diff
@@ -33,41 +42,73 @@ type NetCmd struct {
 // that match the given fabric provider.
 type networkScanCmd struct {
 	logCmd
-	connectedCmd
+	ctlClientCmd
+	jsonOutputCmd
 	FabricProvider string `short:"p" long:"provider" description:"Filter device list to those that support the given OFI provider (default is the provider specified in daos_server.yml)"`
 	AllProviders   bool   `short:"a" long:"all" description:"Specify 'all' to see all devices on all providers.  Overrides --provider"`
 }
 
-func (cmd *networkScanCmd) Execute(args []string) error {
-	var provider string
+func (cmd *networkScanCmd) Execute(_ []string) error {
```

Comment:
The original implementation would, by default, scan the fabric for the devices supporting the provider specified in the daos_server.yml.  I checked out latest build and see that behavior no longer exists.  That was not an intentional change by me.  I'm not sure when that occurred.  The default (on master) is currently a search for devices spanning all providers.  If we keep the existing behavior to scan for all providers unless told otherwise on the command line, then we can delete 'AllProviders' as an option here.  Otherwise, it would make sense to fix the default behavior so it uses the provider in the yaml config by default, and is then overriden by the FabricProvider option or AllProviders option.

Reply:
I think maybe you're thinking of the daos_server version of this command? I haven't changed that: https://github.com/daos-stack/daos/blob/master/src/control/cmd/daos_server/network.go#L63

Reply:
You are right!  I conflated the the daos_server network scan with the dmg network scan.  Sorry for that.